configurablecare/app/src/main/java/com/google/fhir/examples/configurablecare/FhirApplication.kt

```diff
@@ -0,0 +1,120 @@
+import com.google.android.fhir.sync.Sync
+import com.google.android.fhir.sync.remote.HttpLogger
+import com.google.android.fhir.workflow.FhirOperator
+import com.google.android.fhir.workflow.FhirOperatorBuilder
+import com.google.fhir.examples.configurablecare.care.CarePlanManager
+import com.google.fhir.examples.configurablecare.care.ConfigurationManager.getTaskConfigMap
+import com.google.fhir.examples.configurablecare.care.TaskManager
+import com.google.fhir.examples.configurablecare.data.FhirSyncWorker
+import com.google.fhir.examples.configurablecare.external.ValueSetResolver
+import timber.log.Timber
+
+class FhirApplication : Application(), DataCaptureConfig.Provider {
+  private val BASE_URL = "http://10.0.2.2:8088/fhir/"
+  // Only initiate the FhirEngine when used for the first time, not when the app is created.
+  private val fhirEngine: FhirEngine by lazy { constructFhirEngine() }
+  private val knowledgeManager: KnowledgeManager by lazy { constructKnowledgeManager() }
+  private val fhirOperator: FhirOperator by lazy { constructFhirOperator() }
+  private val taskManager: TaskManager by lazy { constructTaskManager() }
+  private val carePlanManager: CarePlanManager by lazy { constructCarePlanManager() }
+  private var dataCaptureConfig: DataCaptureConfig? = null
+
+  private val dataStore by lazy { DemoDataStore(this) }
+
+  override fun onCreate() {
+    super.onCreate()
+    if (BuildConfig.DEBUG) {
+      Timber.plant(Timber.DebugTree())
+    }
+    FhirEngineProvider.init(
+      FhirEngineConfiguration(
+        enableEncryptionIfSupported = false,
+        RECREATE_AT_OPEN,
+        ServerConfiguration(
+          BASE_URL,
+          httpLogger =
+            HttpLogger(
+              HttpLogger.Configuration(
+                if (BuildConfig.DEBUG) HttpLogger.Level.BODY else HttpLogger.Level.BASIC
+              )
+            ) { Timber.tag("App-HttpLog").d(it) }
+        )
+      )
+    )
+    Sync.oneTimeSync<FhirSyncWorker>(this)
+
+    dataCaptureConfig =
+      DataCaptureConfig().apply {
+        urlResolver = ReferenceUrlResolver(this@FhirApplication as Context)
+        valueSetResolverExternal = object : ValueSetResolver() {}
+        xFhirQueryResolver = XFhirQueryResolver { fhirEngine.search(it) }
+      }
+    ValueSetResolver.init(this@FhirApplication)
+  }
+
+  private fun constructFhirEngine(): FhirEngine {
+    return FhirEngineProvider.getInstance(this)
+  }
+
+  private fun constructKnowledgeManager(): KnowledgeManager {
+    return KnowledgeManager.createInMemory(this)
+  }
+  private fun constructFhirOperator(): FhirOperator {
+    return FhirOperatorBuilder(this.applicationContext)
+      .withFhirContext(FhirContext.forR4())
+      .withFhirEngine(fhirEngine)
+      .withIgManager(knowledgeManager)
+      .build()
+  }
+
+  private fun constructCarePlanManager(): CarePlanManager {
+    return CarePlanManager(fhirEngine, knowledgeManager, fhirOperator, taskManager, this)
```

Comment:
We don't need `constructFhirOperator`, `constructKnowledgeManager` functions. These can be constructed here and passed to CarePlanManager.

Also do we really need to construct them here or can it be done in CarePlanManager ?